dialogue-core/src/main/java/com/palantir/dialogue/core/NodeSelectionStrategyChannel.java

```diff
@@ -101,21 +134,68 @@ private static LimitedChannel getUpdatedNodeSelectionStrategy(
                 if (previousNodeSelectionStrategy instanceof PinUntilErrorNodeSelectionStrategyChannel) {
                     PinUntilErrorNodeSelectionStrategyChannel previousPinUntilError =
                             (PinUntilErrorNodeSelectionStrategyChannel) previousNodeSelectionStrategy;
-                    return PinUntilErrorNodeSelectionStrategyChannel.of(
-                            Optional.of(previousPinUntilError.getCurrentChannel()),
-                            updatedStrategy,
-                            channels,
-                            pinuntilerrorMetrics,
-                            random,
-                            channelName);
+                    return channelBuilder
+                            .channel(PinUntilErrorNodeSelectionStrategyChannel.of(
+                                    Optional.of(previousPinUntilError.getCurrentChannel()),
+                                    updatedStrategy,
+                                    channels,
+                                    pinuntilerrorMetrics,
+                                    random,
+                                    channelName))
+                            .build();
                 }
-                return PinUntilErrorNodeSelectionStrategyChannel.of(
-                        Optional.empty(), updatedStrategy, channels, pinuntilerrorMetrics, random, channelName);
-            case ROUND_ROBIN:
+                return channelBuilder
+                        .channel(PinUntilErrorNodeSelectionStrategyChannel.of(
+                                Optional.empty(), updatedStrategy, channels, pinuntilerrorMetrics, random, channelName))
+                        .build();
+            case BALANCED:
                 // When people ask for 'ROUND_ROBIN', they usually just want something to load balance better.
                 // We used to have a naive RoundRobinChannel, then tried RandomSelection and now use this heuristic:
-                return new BalancedNodeSelectionStrategyChannel(channels, random, tick, metrics, channelName);
+                return channelBuilder
+                        .channel(new BalancedNodeSelectionStrategyChannel(channels, random, tick, metrics, channelName))
+                        .build();
+            case UNKNOWN:
         }
         throw new SafeRuntimeException("Unknown NodeSelectionStrategy", SafeArg.of("unknown", updatedStrategy));
```

Comment:
woah if we got an unknown value surely we'd want to just preserve the old thing??? as written doesn't this mean that if a server started recommending `Node-Selection-Strategy: BALANCED2` then all their clients could suddenly die??? this would be a mega P0

Reply:
The logic for picking which node selection strategy to use currently ignores `UNKNOWN` strategy requests. This line is only here to keep errorprone happy